Web Inspector: support starting/stopping recordings from the console in a Worker

dcrousso opened the pull request:
#### c322845a4a00ed5e91f9b00add48d9c1155852b0
<pre>
Web Inspector: support starting/stopping recordings from the console in a Worker
<a href="https://bugs.webkit.org/show_bug.cgi?id=243362">https://bugs.webkit.org/show_bug.cgi?id=243362</a>
<a href="https://rdar.apple.com/98223237">rdar://98223237</a>

Reviewed by BJ Burg.

It&apos;s been possible to capture `&lt;canvas&gt;` recordings in a `Worker` from the UI for a while now (see `WorkerCanvasAgent`) so support it via `console.record()` as well.

* Source/WebCore/workers/WorkerConsoleClient.cpp:
(WebCore::objectArgumentAt): Added.
(WebCore::WorkerConsoleClient::record):
(WebCore::WorkerConsoleClient::recordEnd):

Canonical link: <a href="https://commits.webkit.org/303230@main">https://commits.webkit.org/303230@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/792dfde1f0376749eccceb63f15e65b81a217d09

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/130198 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/2469 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/41152 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/137611 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/81754 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/132069 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/2453 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/2359 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/99187 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/67032 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/133145 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/1834 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/116609 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/79879 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/1753 "Passed tests") | [⏳ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/macOS-Tahoe-Debug-API-Tests-EWS "Waiting to run tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/80872 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/110282 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/34737 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/140087 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/2259 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/35244 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/107710 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/2303 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/2106 "Found 1 new test failure: imported/w3c/web-platform-tests/webvtt/rendering/cues-with-video/processing-model/selectors/cue/white-space_normal_wrapped.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/107596 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27666 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/1792 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/112952 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/55202 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/2329 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/31402 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/2146 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/65716 "Built successfully") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/2350 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/2255 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rcaliman-apple reviewed (commented):
This looks identical to the implementation in `FrameConsoleClient` (renamed from `PageConsoleClient`) so I guess it's fine.

LGTM 👍 @burg, can it get a R+?

https://github.com/WebKit/WebKit/blob/f8af49f640cc922f066d0e349ff119a5ea6a7eb2/Source/WebCore/page/FrameConsoleClient.cpp#L330-L350

~~I'm curious where the existing implementation is used in the UI. For example, the only call site I can find for `recordEnd` is in [`WI.Recording.prototype.toHTML()`](https://github.com/WebKit/WebKit/blob/f8af49f640cc922f066d0e349ff119a5ea6a7eb2/Source/WebInspectorUI/UserInterface/Models/Recording.js#L742) introduced by https://bugs.webkit.org/show_bug.cgi?id=195311~~

~~I _can_ see it [exposed](https://github.com/WebKit/WebKit/blob/f8af49f640cc922f066d0e349ff119a5ea6a7eb2/Source/JavaScriptCore/runtime/ConsoleObject.cpp#L117) on the `console` object for calling from the Console Prompt.~~

**Edit:** Scratch that. I see what you mean. This is calling into `InspectorCanvasAgent::consoleStartRecordingCanvas()`

dcrousso committed:
Web Inspector: support starting/stopping recordings from the console in a Worker
https://bugs.webkit.org/show_bug.cgi?id=243362
rdar://98223237

Reviewed by BJ Burg.

It's been possible to capture `<canvas>` recordings in a `Worker` from the UI for a while now (see `WorkerCanvasAgent`) so support it via `console.record()` as well.

* Source/WebCore/workers/WorkerConsoleClient.cpp:
(WebCore::objectArgumentAt): Added.
(WebCore::WorkerConsoleClient::record):
(WebCore::WorkerConsoleClient::recordEnd):

Canonical link: https://commits.webkit.org/303230@main
